web/client/utils/LayersUtils.js

```diff
@@ -707,7 +707,8 @@ export const saveLayer = (layer) => {
         legendOptions: layer.legendOptions,
         tileSize: layer.tileSize,
         version: layer.version,
-        expanded: layer.expanded || false
+        expanded: layer.expanded || false,
+        ...(layer.visibilityLimitType ? { visibilityLimitType: layer.visibilityLimitType } : undefined)
```

Comment:
Here I meant, why not ?

```suggestion
       visibilityLimitType: layer.visibilityLimitType
```

Reply:
Tests were failing because the functionality checks the data that is effectively saved to be the one you want.
You can keep this implementation, but add a test that if visibilityLimitType is passed, then is persisted, otherwise it is not (it is a temporary variable, so it makes sense to have it.

Reply:
So if we don't need to persist visibilityLimitType then this is unnecessary. So removing it. @offtherailz